### PR TITLE
feat: api /workspaces/<str:slug>/issues/<str:project__identifier>-<str:issue__identifier>/

### PR DESCRIPTION
**Pull request by MichaelErmer**
This PR adds an API endpoint to find an issue by its sequenceId (XYZ-1234).

This is required to connect plane to any other services, where the tickets are only referenced by their sequenceId.

### CODE REVIEW

**Comment by MichaelErmer** [comment]
Also fixed #4171

**Comment by ryandevz** [comment]
Highly awaiting this feature.

**Comment by MichaelErmer** [comment]
> Hi @MichaelErmer, thanks for contributing to Plane.
> 
> Since the url does not contain project_id the project level permissions can't be validated. The issue should not be available to a user who is not part of the project.

The Project is resolved from the Project SLUG which is part of the Ticket Identifier XYZ = project, 1234 = ticket sequence

**Comment by MichaelErmer** [comment]
@pablohashescobar  is there anything you want changed / tested? Id happily add a test if there is a setup to do so.

**Comment by pablohashescobar** [comment]
Hi @MichaelErmer 
> Project is resolved from the Project SLUG which is part of the Ticket Identifier XYZ = project, 1234 = ticket sequence

You are correct but the permission's are not validated that is if the user has access to that project or not. Can you add that layer as well.

**Comment by MichaelErmer** [comment]
@pablohashescobar I modified the permission check to accept project__identifier instead of project_id, it works as expected, I tested it with 2 projects and access to only one.
image
image

File: apiserver/plane/api/urls/issue.py [hunk 1]
```diff
@@ -6,9 +6,15 @@
     IssueLinkAPIEndpoint,
     IssueCommentAPIEndpoint,
     IssueActivityAPIEndpoint,
+    WorkspaceIssueAPIEndpoint,
 )
 
 urlpatterns = [
+    path(
+        "workspaces/<str:slug>/issues/<str:project__identifier>-<str:issue__identifier>",
```
Comment (pablohashescobar) [inline apiserver/plane/api/urls/issue.py:14]: @MichaelErmer can you please add a trailing `/` here.

**Review by pablohashescobar — CHANGES_REQUESTED** [review]

File: apiserver/plane/api/urls/issue.py (hunk 1 above)
Comment (MichaelErmer) [reply apiserver/plane/api/urls/issue.py:14]: done

File: apiserver/plane/api/views/issue.py [hunk 2]
```diff
@@ -51,6 +52,65 @@
...
+            .select_related("workspace")
+            .select_related("state")
+            .select_related("parent")
+            .prefetch_related("assignees")
+            .prefetch_related("labels")
+            .order_by(self.kwargs.get("order_by", "-created_at"))
+        ).distinct()
+
+    def get(self, request, slug, project__identifier=None, issue__identifier=None):
+        if issue__identifier and project__identifier:
```
Comment (pablohashescobar) [inline apiserver/plane/api/views/issue.py:96]: @MichaelErmer can you also add an `else` case here.

**Review by pablohashescobar — CHANGES_REQUESTED** [review]

File: apiserver/plane/api/views/issue.py (hunk 2 above)
Comment (pablohashescobar) [reply apiserver/plane/api/views/issue.py:96]: Rest LGTM.

Comment (MichaelErmer) [reply apiserver/plane/api/views/issue.py:96]: the else is covered by django, it will 404, what else should it do?

**Comment by MichaelErmer** [comment]
@pablohashescobar can you have a look again? The else is handled by django, it 404's what else would you like to happen?

**Comment by pablohashescobar** [comment]
> @pablohashescobar can you have a look again? The else is handled by django, it 404's what else would you like to happen?

@MichaelErmer, got it. The PR looks good. We will merge it. Thank you for contributing to Plane.

**Comment by MichaelErmer** [comment]
> > @pablohashescobar can you have a look again? The else is handled by django, it 404's what else would you like to happen?
> 
> 
> 
> @MichaelErmer, got it.  We can explicitly handle that condition returning a JSON response. 
> 
> 
> 
> Also can you please make a small change can you change the URL to `workspaces/<str:slug>/projects/<str:project__identifier>/issues/<str:issue__identifier>/`
> 
> This will make it uniform to all the other APIs.
> 
> 
> 
> Sorry for the repeated pings—just aiming to align everything with our project standards. Really appreciate your patience and effort on this!

It is intentional to have a url that does resolve the project, the project is part of the identifiers XYZ-1234, XYZ resolves the project. 

This is to be used by an integration that only needs the identifier and can then use the API to perform updates/actions on plane. There can be multiple projects and using your approach would then require to make all projects known to the external part and do multiple lookups until one matches...

**Comment by pablohashescobar** [comment]
@MichaelErmer, correct makes sense. Thank you for pointing out.

**Review by pablohashescobar — APPROVED** [review]

**Pull request merged**